server/services/omnichannel-voip/service.ts

```diff
@@ -72,4 +155,156 @@ export class OmnichannelVoipService extends ServiceClass implements IOmnichannel
 			})),
 		};
 	}
+
+	/* Voip calls */
+	async findAgent(agentId: string): Promise<any> {
+		return this.normalizeAgent(agentId);
+	}
+
+	async getNewRoom(guest: ILivechatVisitor, agent: any, rid: string, roomInfo: any): Promise<IOmnichannelVoipServiceResult> {
+		this.logger.debug(`Attempting to find or create a room for visitor ${guest._id}`);
+		const message = {
+			_id: Random.id(),
+			rid,
+			msg: '',
+			token: guest.token,
+			ts: new Date(),
+		};
+
+		let room = await this.voipRoom.findOneById(message.rid);
+		let newRoom = false;
+		if (room && !room.open) {
+			this.logger.debug(`Last room for visitor ${guest._id} closed. Creating new one`);
+			message.rid = Random.id();
+			room = null;
+		}
+		if (room == null) {
+			// delegate room creation to QueueManager
+			this.logger.debug(`Calling QueueManager to request a room for visitor ${guest._id}`);
+			// room = await QueueManager.requestRoom({ guest, message, roomInfo, agent: defaultAgent, extraData });
+			const name = roomInfo?.fname || guest.name || guest.username;
+			const roomId = await this.createVoipRoom(rid, name, agent, guest);
+			room = await this.voipRoom.findOneVoipRoomById(roomId);
+			newRoom = true;
+			this.logger.debug(`Room obtained for visitor ${guest._id} -> ${room?._id}`);
+		}
+		if (!room) {
+			this.logger.debug(`Visitor ${guest._id} trying to access another visitor's room`);
+			throw new Meteor.Error('cannot-access-room');
+		}
+		return {
+			result: {
+				room,
+				newRoom,
+			},
+		};
+	}
+
+	async findRoom(token: string, rid: string): Promise<IOmnichannelVoipServiceResult> {
+		const fields = {
+			t: 1,
+			departmentId: 1,
+			servedBy: 1,
+			open: 1,
+			v: 1,
+			ts: 1,
+		};
+		let room: IVoipRoom | null;
+		if (!rid) {
+			room = await this.voipRoom.findOneByVisitorToken(token, fields);
+		} else {
+			room = await this.voipRoom.findOneByIdAndVisitorToken(rid, token, fields);
+		}
+		return Promise.resolve({
+			result: room,
+		});
```

Comment:
this is good example how to not use promises, instead try something like: 
```suggestion
		if (!rid) {
			return { result: await this.voipRoom.findOneByVisitorToken(token, fields) };
		} 
		return  { result: await this.voipRoom.findOneByIdAndVisitorToken(rid, token, fields) };
		}
```

or ```suggestion
		return { result: await (rid ? this.voipRoom.findOneByIdAndVisitorToken(rid, token, fields):  this.voipRoom.findOneByVisitorToken(token, fields)) };
```
btw do we really need this `result` property ? why not only return the room instead

Reply:
Hello Guilherme Thanks for reviewing the code. :)
Fixed the promise part of the code.
This was in voip/service.ts to avoid changing the return type of executeCommand function (Which can pull multiple types of information from the connector) for every newly supported object. I copied the same pattern for uniformity here.